src/OrchardCore.Modules/OrchardCore.Media/Processing/MediaTokenService.cs

```diff
@@ -80,12 +75,50 @@ public string AddTokenToPath(string path)
                 }
             }
 
-            return QueryHelpers.AddQueryString(path.Substring(0, pathIndex), processingCommands);
+            return AddQueryString(path.AsSpan(0, pathIndex), processingCommands);
+        }
+
+        private void ParseQuery(
```

Comment:
same as in `QueryHelpers`, but produces two dictionaries based on whether known commands matches